PREDICTION_FOR_REPETITION

```diff
@@ -0,0 +1,66 @@
+package ArrayDataPrediction;
+
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Scanner;
+import java.util.Set;
+import GenerateData.DataGenerator;
+import GenerateData.Initiator;
+
+
+public class CheckForRepetetion {
+	public static void main(String[] args) {
+		int arraysize;	
+    
+		Scanner InputSize=new Scanner(System.in);
+		System.out.println("Enter the size of the array: ");
+		arraysize= InputSize.nextInt();
+		
+		DataGenerator dataGenerator = new DataGenerator(DataGenerator.PATTERN_REPEATED, arraysize);
+    System.out.println(dataGenerator.generate());
+    List<Integer> GetArray=dataGenerator.generate();
+    Integer[] TArr=GetArray.toArray(new Integer[GetArray.size()]);
+		int length=TArr.length;
+		TestReps(TArr, length);
+	}
+	
+
+	public static void TestReps(Integer TArr[],int size) {
+		
+		float Threshold;
+		float count=0;
+		float Percentage;
+		int length=size;
+		Scanner ThresholdInput= new Scanner(System.in);
+		System.out.print("Enter the threshold percentage:");
+		Threshold= ThresholdInput.nextFloat()/100;
+		
+		long begin = System.currentTimeMillis();
+		System.out.println("Threshold is set to : "+Threshold*100+"%");
+		System.out.println("Size of array :"+length);
+		
+		Set<Integer> mySet= new HashSet<Integer>();
+		
+		for(Integer ArrayElements: TArr) {
+			if (mySet.add(ArrayElements)==false) {
+				count++;				
+			}
+			
+		}
+		
+		Percentage=(count/length)*100;
+		
+		System.out.println("Total repetitions :"+count);
+		System.out.println("Percentage of repetitions: "+Percentage+"%");
+		
+		if(Percentage>Threshold*100) 
```

Comment:
We shouldn't be getting the threshold from the users. We must be able to set a threshold based on array pattern.

Reply:
Yes! we are not going to get the 'threshold' value from the end user. Obviously, the user doesn't have to know about anything but the array he is trying to sort.  Here the threshold (and the size of the array) is used by us to check various test cases and decide the threshold range where the sorting algorithms, that are good for repetition arrays, work best. [I would request you to please read the word file also, where the working is explained.] :)